src/catch2/internal/catch_console_colour.cpp

```diff
@@ -185,7 +223,10 @@ namespace {
 #if defined( CATCH_INTERNAL_HAS_ISATTY ) && \
     !( defined( __DJGPP__ ) && defined( __STRICT_ANSI__ ) )
             ErrnoGuard _; // for isatty
-            useColour = useColour && isatty( STDOUT_FILENO );
+            useColour = useColour && _isatty( _fileno( stdout ) );
```

Comment:
Hate that part... That check should be done in `CoutStream` / `CerrStream`. The whole `_isatty` stuff is duplicated, and it's not as if `isConsole()` ***needs*** to be lying.

Not to mention this is making a judgment on `stdout` even if `stderr` happens to be in use.